1. extraction/gnaf.py

```diff
@@ -0,0 +1,116 @@
+import pandas as pd
+import pickle
+import os
+
+# Accessing the DataFiles Path
+home_dir = os.path.expanduser("~")
+datafiles_path = os.path.join(home_dir, "Desktop", "DataFiles")
+
+# Setting the FilesIn as the source files, and the AppStaging for the pickles
+in_path = os.path.join(datafiles_path, "FilesIn")
+standard_in = os.path.join(in_path, "Standard")
+authority_code_in = os.path.join(in_path, "Authority Code")
+pickle_file_path = os.path.join(datafiles_path, "AppStaging", 'locational-data.pkl')
+files_out_path = os.path.join(datafiles_path, "FilesOut")
+
+def read_codes():
+    # Initialize an empty dictionary to hold the DataFrames
+    dataframes_dict = {}
+
+    # Loop through all the files in codes
+    for filename in os.listdir(authority_code_in):
+        if filename.endswith('.psv'):
+            file_path = os.path.join(authority_code_in, filename)
+            
+            try:
+                # Read the PSV file into a DataFrame
+                df = pd.read_csv(file_path, delimiter='|')
+                
+                # Use the filename (without extension) as the key for the dictionary
+                key = os.path.splitext(filename)[0]
+                dataframes_dict[key] = df
+                
+            # The code files may be empty - the version I had were empty
+            except pd.errors.EmptyDataError:
+                continue
+            except pd.errors.ParserError:
+                print(f"Skipped malformed file: {filename}")
+            except Exception as e:
+                print(f"Error processing file {filename}: {e}")
```

Comment:
use `logger.warning()` instead of `print`, also you can just wrap the function in `@logger.catch()` to catch and warn about the exceptions without terminating